### PR TITLE
Performance optimisation for carddav backend blob data read

### PR DESCRIPTION
**Pull request by miaulalala**
Remove unneccessary looping and duplication of array values in CardDav Backend method when parsing blob data to remove photos.

### CODE REVIEW

File: apps/dav/lib/CardDAV/CardDavBackend.php [hunk 1]
```diff
@@ -933,11 +933,17 @@ private function readBlob($cardData, &$modified = false) {
 			$cardData = stream_get_contents($cardData);
 		}
 
-		$cardDataArray = explode("\r\n", $cardData);
+		// Micro optimisation
+		// don't loop through
+		if(strpos($cardData, 'PHOTO') === 0) {
```
Comment (nickvergessen) [inline apps/dav/lib/CardDAV/CardDavBackend.php:938]:
```suggestion
		if (strpos($cardData, 'PHOTO') === 0) {
```

File: apps/dav/lib/CardDAV/CardDavBackend.php [hunk 2]
```diff
@@ -933,11 +933,17 @@ private function readBlob($cardData, &$modified = false) {
 			$cardData = stream_get_contents($cardData);
 		}
 
-		$cardDataArray = explode("\r\n", $cardData);
+		// Micro optimisation
+		// don't loop through
+		if (strpos($cardData, 'PHOTO') === 0) {
```
Comment (juliushaertl) [inline apps/dav/lib/CardDAV/CardDavBackend.php:938]: To micro optimize a bit more maybe:

```suggestion
		if (strpos($cardData, 'PHOTO:data:image/') === 0) {
```

**Review by nickvergessen — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]
:elephant:

**Review by tcitworld — APPROVED** [review]

**Comment by miaulalala** [comment]
Performance check is shot (maintenance mode) and Drone has a Samba failure.

**Pull request merged**